controller/usercontroller.js

```diff
@@ -0,0 +1,349 @@
+//         if(isMatch){
+//             const userid = finduser._id;
+//             let token = jwt.sign({userid,email},process.env.JWT_ACCESS_TOKEN,{expiresIn:process.env.EXPIRY})
+//             const tokengen = new AccessToken({
+//                 userid:userid,
+//                 accesstoken:token
+//             });
+//             tokengen.save();
+//             res.header('jwt',tokengen.accesstoken)
+//             res.status(201).json({
+//                 message:"USer found"
+//             })
+//         }
+//         else{
+//             res.status(500).json({message:"password does not matched please try again"})
+//         }
+//     } catch (error) {
+//         res.status(500).json({message:"something went wronng"})
+//     }
+// }
+// login and generaating random number;
+// const loginandgeneratingnumber =async(req,res)=>{
+//     try {
+//                 const {email, password} = req.body;
+//                 const finduser = await UserSignup.findOne({email:email}).lean()
+//                 if(!finduser) return res.status(500).json({message:"USer not found please login"});
+//                 const isMatch = await bcrypt.compare(password,finduser.password);
+//                 if(isMatch){
+//                     const userid = finduser._id
+//                     var today = new Date();
+//                     today = (today.getTime()+3600);
+//                     const tokendetail= new AccessToken({
+//                         userid:userid,
+//                         accesstoken:md5(userid),
+//                         expiry:today
+//                     })
+//                     tokendetail.save();
+//                     let token = finduser._id
+//                     res.header('authorization',tokendetail.accesstoken);
+//                     res.status(200).json({
+//                         message:"User found",
+//                         data:token
+//                     })
+//                 }
+//                 else{
+//                     res.status(500).json({message:"something went wrong"})
+//                 }
+//             } catch (error) {
+//                 res.status.json({message:"something went wronng"})
+//             }
+//         }
+
+
+//get user
+const getuser = (req, res) => {
+    let token = req.data;
+
+    res.status(200).json({
+        message: "you are authenticated",
+        "userdetail": token
+    });
+};
+
+//delete user
+
+const deleteuser = async (req, res) => {
+    let token = req.id;
+    try {
+        const deleted = await UserSignup.findByIdAndDelete({ _id: token })
+        res.status(200).json({ message: "Data deleted" })
+
+
+
+    } catch (error) {
+        res.status(500).json({ message: "Internal server error" })
+        console.log(error)
+    }
+
+};
+
+//pagination
+const pagination = async (req, res) => {
+    try {
+        let page = req.params.page;
+        let size = 10;
```

Comment:
size shouldn't be hardcoded

Reply:
should we take it from user or I put this in env